File: LeetCode/Easy/1491 - Average Salary Excluding the Minimum and Maximum Salary.js
/**
 * @param {number[]} salary
 * @return {number}
 */
var average = function(salary) {
    const n = salary.length;
    let min = Infinity, max = -Infinity, sum = 0;
    for (let i = 0; i < n; ++i) {
        sum += salary[i];
        min = Math.min(min, salary[i]);
        max = Math.max(max, salary[i]);
    }
    return (sum - max - min) / (n - 2);
};

File: LeetCode/Easy/1491 - Average Salary Excluding the Minimum and Maximum Salary/1491 - Average Salary Excluding the Minimum and Maximum Salary.cpp
class Solution {
public:
    double average(vector<int>& salary) {
        int mn = INT_MAX, mx = INT_MIN, sum = 0;
        for (auto s : salary) {
            sum += s;
            mn = min(mn, s);
            mx = max(mx, s);
        }
        return (double)(sum - mx - mn) / (salary.size() - 2);
    }
};